src/lib/logical_query_plan/union_node.cpp

```diff
@@ -45,6 +45,26 @@ bool UnionNode::is_column_nullable(const ColumnID column_id) const {
   return left_input()->is_column_nullable(column_id) || right_input()->is_column_nullable(column_id);
 }
 
+std::shared_ptr<LQPUniqueConstraints> UnionNode::unique_constraints() const {
+  switch (set_operation_mode) {
+    case SetOperationMode::Unique:
+    case SetOperationMode::Positions:
+      // UnionPositions merges two reference tables with the same original table(s). Any duplicate RowIDs are
+      // filtered out. As a consequence, existing unique constraints from input tables can be forwarded.
+      Assert(*left_input()->unique_constraints() == *right_input()->unique_constraints(),
+             "Input tables should have the same constraints.");
+      return _forward_unique_constraints();
+
+    case SetOperationMode::All:
+      // With UnionAll two tables of the same schema become merged. The resulting table might contain duplicates.
```

Comment:
"of the same schema' - not necessarily.

Reply:
You are right. Thanks!

Changed to "With UnionAll, two tables become merged. The resulting table might contain duplicates."